src/Symfony/Component/Cache/Adapter/AbstractAdapter.php

```diff
@@ -0,0 +1,227 @@
+            CacheItem::class
+        );
+        $this->mergeByLifetime = \Closure::bind(
+            function ($deferred, $namespace) {
+                $byLifetime = array();
+
+                foreach ($deferred as $key => $item) {
+                    if (0 <= $item->lifetime) {
+                        $byLifetime[$item->lifetime][$namespace.$key] = $item->value;
+                    }
+                }
+
+                return $byLifetime;
+            },
+            $this,
+            CacheItem::class
+        );
+    }
+
+    /**
+     * Fetches several cache items.
+     *
+     * @param array $ids The cache identifiers to fetch.
+     *
+     * @return array The corresponding values found in the cache.
+     */
+    abstract protected function doFetch(array $ids);
+
+    /**
+     * Persists several cache items immediately.
+     *
+     * @param array $values   The values to cache, indexed by their cache identifier.
+     * @param int   $lifetime The lifetime of the cached values, 0 for persisting until manual cleaning.
+     *
+     * @return array|bool The identifiers that failed to be cached or a boolean stating if caching succeeded or not.
+     */
+    abstract protected function doSave(array $values, $lifetime);
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getItem($key)
+    {
+        $id = $this->getId($key);
+
+        if ($this->deferred) {
+            $this->commit();
+        }
+        if (isset($this->deferred[$key])) {
+            return $this->deferred[$key];
+        }
+
+        $f = $this->createCacheItem;
+        $isHit = false;
+        $value = null;
+
+        foreach ($this->doFetch(array($id)) as $value) {
+            $isHit = true;
+        }
+
+        return $f($key, $value, $isHit);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getItems(array $keys = array())
+    {
+        if ($this->deferred) {
+            $this->commit();
+        }
+        $f = $this->createCacheItem;
+        $ids = array();
+        $items = array();
+
+        foreach ($keys as $key) {
+            $id = $this->getId($key);
+
+            if (isset($this->deferred[$key])) {
+                $items[$key] = $this->deferred[$key];
+            } else {
+                $ids[$key] = $id;
+            }
+        }
+
+        $values = $this->doFetch($ids);
+
+        foreach ($ids as $key => $id) {
+            $isHit = isset($values[$id]);
+            $items[$key] = $f($key, $isHit ? $values[$id] : null, $isHit);
+        }
+
+        return $items;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function deleteItems(array $keys)
+    {
+        $ok = true;
+
+        foreach ($keys as $key) {
+            $ok = $this->deleteItem($key) && $ok;
+            unset($this->deferred[$key]);
+        }
+
+        return $ok;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function save(CacheItemInterface $item)
+    {
+        if (!$item instanceof CacheItem) {
```

Comment:
Why do you need to do this?
It breaks the usage of `CacheItemInterface` interface used as type hinting.